src/compiler/core.ts

```diff
@@ -149,16 +149,17 @@ namespace ts {
      * Iterates through `array` by index and performs the callback on each element of array until the callback
      * returns a falsey value, then returns false.
      * If no such value is found, the callback is applied to each element of array and `true` is returned.
+     * If `array` is undefined, return false.
      */
     export function every<T>(array: readonly T[] | undefined, callback: (element: T, index: number) => boolean): boolean {
-        if (array) {
-            for (let i = 0; i < array.length; i++) {
-                if (!callback(array[i], i)) {
-                    return false;
-                }
+        if (!array) {

```

Comment:
interesting this should be !length(array) instead

Reply:
A little confused, another review says "revert changes in core", so this is also reverted?